openjdk/mmtkUpcalls.cpp

```diff
@@ -65,10 +65,12 @@ static void mmtk_stop_all_mutators(void *tls, void (*create_stack_scan_work)(voi
     }
   }
   log_debug(gc)("Finished enumerating threads.");
+  nmethod::oops_do_marking_prologue();
 }
 
 static void mmtk_resume_mutators(void *tls) {
-  ClassLoaderDataGraph::purge();
+  nmethod::oops_do_marking_epilogue();
```

Comment:
Is is correct to call the epilogue after the GC instead of after root scanning? I honestly don't have the knowledge to understand what the functions are doing, so am unsure if it is considered correct/okay to call the epilogue after the GC has occurred.

Reply:
Looks like this function only prunes an internal nmethod list. I guess it should work correctly as long as all the necessary nmethods are scanned properly.